### DISCUSSION TITLE
Compiling with the same settings & features plus one additional feature - Athom PG05V2

### DISCUSSION THREAD

**Discussion by scottmeup** [arendst/Tasmota#22202] [Support request]
Hi, I'd like to build a version of the Tasmota firmware that came pre-flashed on a Athom PG05V2-AU10A smart plug, and add the `USE_PING` feature. I'd like to keep the same dashboard, default settings, and enabled features apart from the addition of `USE_PING`.

I'm planning to use [TasmoCompiler](https://github.com/benzino77/tasmocompiler) mostly due to my level of skill and familiarity with compiling manually.

I'd like to know:

1. Is there a way to check which settings and features are already enabled on the device?
2. Can the information from question 1 be easily translated into TasmoCompiler configuration options?
3. Is the [configuration yaml](https://github.com/athom-tech/athom-configs/blob/main/athom-smart-plug-v2.yaml) for the device from Athom required or helpful? If so, how should it be used?
4. To add the `USE_PING` feature with TasmoCompiler, is the following code in the Custom Parameters field correct?
```
#ifndef USE_PING
#define USE_PING
#endif
```

The information about the PG05V2-AU10A I've found so far is a template and specifications on the Athom Site, as well as a yaml config file on the Athom github repo.
- https://www.athom.tech/blank-1/au-plug
- https://github.com/athom-tech/athom-configs/blob/main/athom-smart-plug-v2.yaml

&nbsp;

Any suggestions or pointers on how to do this are appreciated.

Thanks in advance.

**Comment by sfromis** [comment]
1. The tool https://github.com/tasmota/decode-status/ can extract configuration in a few formats, and https://github.com/tasmota/decode-config/ gives a shorter list of features and setoption values. You can also check what gpio config is in the template, which I'd expect to match https://templates.blakadder.com/athom_PG05V2-AU16A-TAS.html (same pin config for different internationalized variants).
2. I'd expect all of this to not be needed, as default Tasmota firmware should work on the plug without amendments (also when adding new features like Ping), and the device-specific configuration will be kept after Tasmota firmware upgrade. In case of changing defaults for values you can configure from inside Tasmota, those defaults will not be used unless you do a full reset.
3. That ESPHome config is not directly useful for Tasmota, except that it can be a place to gather details about a possible setup, but when you already have a working Tasmota setup....
4. Looks ok
